routing.rst

```diff
@@ -290,23 +327,295 @@ with a locale. This can be done by defining a different prefix for each locale
+         */
+        public function show(Post $post)
+        {
+            // ...
+        }
+    }
+
+In this example, the route of the ``index()`` action will be called ``blog_index``
+and its URL will be ``/blog/``. The route of the ``show()`` action will be called
+``blog_post`` and its URL will be ``/blog/{_locale}/posts/{slug}``. Both routes
+will also validate that the ``_locale`` parameter matches the regular expression
+defined in the class annotation.
+
+The same can be used when importing routes from configuration files:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/routes/annotations.yaml
+        controllers:
+            resource: '../src/Controller/'
+            type: annotation
+            # this is added to the beginning of all imported route URLs
+            prefix: '/blog'
+            # this is added to the beginning of all imported route names
+            name_prefix: 'blog_'
+            # these requirements are added to all imported routes
+            requirements:
+                locale: 'en|es|fr'
+
+    .. code-block:: xml
+
+        <!-- config/routes/annotations.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <import resource="../src/Controller/" type="annotation">
+                <!-- this is added to the beginning of all imported route URLs -->
+                <prefix>/blog</prefix>
+                <!-- this is added to the beginning of all imported route names -->
+                <name-prefix>blog_</name-prefix>
+                <!-- these requirements are added to all imported routes -->
+                <requirement key="locale">en|es|fr</requirement>
+            </import>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes/annotations.php
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->import('../src/Controller/', 'annotation')
+                // this is added to the beginning of all imported route URLs
+                ->prefix('/blog')
+                // this is added to the beginning of all imported route names
+                ->namePrefix('blog_')
+                // these requirements are added to all imported routes
+                ->requirements(['locale' => 'en|es|fr'])
+            ;
+        };
+
+.. note::
+
+    If any of the prefixed routes defines an empty path, Symfony adds a trailing
+    slash to it. In the previous example, an empty path prefixed with ``/site``
+    will result in the ``/site/`` URL. If you want to avoid this behavior, set
+    the ``trailing_slash_on_root`` option to ``false``:
```

Comment:
Does this also apply to annotations?

Could we include this in the code-block above, commented out with the explanation:

```
# An empty route path ("") will become "/blog/".
# Uncomment this option to make that URL "/blog" instead
# trailing_slash_on_root: false
```

Reply:
It doesn't apply to annotations ... but I love your suggestion to inline this as comments. I love it!